File: week-2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{ // get text
    string text = get_string("text: ");
    // start counter
    int letters = 0, words = 1, sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == ' ')
        {
            words++;
        }
        else if (text[i] == '.' || text[i] == '!' || text[i] == '?')

        {
            sentences++;
        }
    }

    // calculate l and s
    float L = (float) letters / words * 100;
    float S = (float) sentences / words * 100;
    // calculate
    float index = 0.0588 * L - 0.296 * S - 15.8;
    // level
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", (int) round(index));
    }
}

File: week-6/readability/readability.py
def get_string(prompt):
    return input(prompt)


text = get_string("text: ")

counter_letter = 0
counter_words = 0
counter_sentences = 0

for char in text:
    if char.isalpha():
        counter_letter += 1

counter_words = len(text.split())

counter_sentences = text.count('.') + text.count('!') + text.count('?')

L = (counter_letter / counter_words) * 100
S = (counter_sentences / counter_words) * 100
index = 0.0588 * L - 0.296 * S - 15.8

if index >= 16:
    print("Grade 16+")
elif index < 1:
    print("Before Grade 1")
else:
    print(f"Grade {int(round(index))}")
